src/index/cagra/cagra.cu

```diff
@@ -189,11 +168,9 @@ class CagraIndexNode : public IndexNode {
         is.read((char*)(&this->counts_), sizeof(this->counts_));
         this->devs_.resize(1);
         is.read((char*)(&this->devs_[0]), sizeof(this->devs_[0]));
-        auto scoped_device = detail::device_setter{devs_[0]};
+        auto scoped_device = raft_utils::device_setter{devs_[0]};
 
-        thread_local rmm::mr::cuda_memory_resource mr;
-        thread_local rmm::cuda_stream stream;
-        thread_local raft::device_resources res(stream.view(), nullptr, &mr);
+        auto& res = raft_utils::get_raft_resources();
```

Comment:
I would prefer not to use the explicit memory pool except for search. milvus is a distributed system, some nodes are only responsible for build index, they are not performance sensitive tasks to set up explicit pools, which sometimes cause deployment problems. For example, it is necessary to deploy build node and search node on the same card. Both processes will open two gpu memory pools at the same time, and if there is not enough gpu memory it will cause the deployment to fail.

Reply:
We already give users control over the starting and maximum pool size, and the default starting size is quite modest. If the separate pools grow to the point where an allocation fails, it would fail regardless of whether a pool is being used or not.

Basically, what I'm trying to say is that this is an issue regardless of whether or not a memory pool is used for search and index building or just search. In both situations, there's not enough memory, and there are other advantages to using a single unified memory allocator. In fact, it may make it easier to handle situations that push the available hardware to its memory limit by allowing for cleaner failures if we run out of memory.

Reply:
@wphicks 
Thank you for your reply.
I can understand you your opinion. Often we are not very sensitive to build index performance. The problem with the memory pool is that once the memory inited or expands, it doesn't shrink back again. If it is a distributed scenario, buildnode and searchnode are two linux processes running on two separate containers, The two processes are two separate pools of gpu memory pool. it doesn't feel right to dedicate a section of gpu memory to a task that is not called very often.

Reply:
Ah, I understand! How would we feel about allowing the user to optionally (e.g. via an environment variable) disable the memory pool altogether for a deployment? That would keep the case where search and build happen in the same process clean, but it would allow it to be disabled for the case of a dedicated build node. That might also better accommodate situations when search traffic is very "bursty" and we do not want to hang onto device memory corresponding to the peak.